src/Microsoft.Sbom.Api/Entities/output/ValidationResultGenerator.cs

```diff
@@ -63,16 +65,34 @@ public ValidationResultGenerator WithValidationResults(IList<FileValidationResul
         return this;
     }
 
+    public ValidationResultGenerator WithOptionalValidationResults(IList<FileValidationResult> optionalNodeValidationResults)
+    {
+        OptionalNodeValidationResults = optionalNodeValidationResults ?? new List<FileValidationResult>();
+        return this;
+    }
+
     /// <summary>
     /// Finalizes the validation generation and returns a new <see cref="ValidationResult"/> object.
     /// </summary>
     /// <returns></returns>
     public ValidationResult Build()
     {
         List<FileValidationResult> validationErrors;
+        List<FileValidationResult> optionalValidationErrors;
         List<FileValidationResult> skippedErrors;
 
         validationErrors = NodeValidationResults.Where(r => !Constants.SkipFailureReportingForErrors.Contains(r.ErrorType)).ToList();
+
+        // Put all the errors into a list for the output to handle, if there are none then create an empty list
+        if (OptionalNodeValidationResults != null)
+        {
```

Comment:
Why we need to store the optional validation results in a different list?

Reply:
At first I thought this was necessary but this question made me think it over and realized there was a simple change to get rid of the need for it. I updated the PR accordingly.